### PR TITLE
finish ticket "TextField"

### PR DESCRIPTION
**Pull request by maverick-go**
## Ticket

- [TextField](https://www.notion.so/10c3ebb964c04e1991c7592924313ae6?v=30546a829e844023814e1711a74737b2&p=c3ea90f7d5e74cc580d096aef6b218a7&pm=s)

## Screenshots

- image
- image
- image

## Videos

- 
https://github.com/krish-do-goldenowl/training_minhnhat_dart_challenge/assets/139732686/2ce7fab1-c894-4208-a134-343a0942ab76

### CODE REVIEW

**Comment by krish-do-goldenowl** [comment]
@maverick-go you must set the input field on top of the keyboard

image

File: gallery_widgets/lib/features/text_field/widgets/settings_view.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import 'package:gallery_widgets/widgets/setting_row.dart';
+
+class SettingsView extends StatelessWidget {
+  const SettingsView({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 20, horizontal: 10),
+      child: BlocBuilder<TextFieldCubit, TextFieldState>(
```
Comment (krish-do-goldenowl) [inline gallery_widgets/lib/features/text_field/widgets/settings_view.dart]: break out the `BlocBuilder` and add the `buildWhen`

File: gallery_widgets/lib/features/text_field/widgets/text_field_2_views.dart [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+              buildTextField(hasBorder: true, state: state),
+            ],
+          );
+        },
+      ),
+    );
+  }
+
+  Widget buildTextField(
+      {bool hasBorder = false, required TextFieldState state}) {
```
Comment (krish-do-goldenowl) [inline gallery_widgets/lib/features/text_field/widgets/text_field_2_views.dart]: You should not use the state like this => use BlocBuilder for every widget using the state

**Review by krish-do-goldenowl — APPROVED** [review]

**Comment by maverick-go** [comment]
> @maverick-go you must set the input field on top of the keyboard
> 
> image

done

**Pull request merged**